docs/dev/commitlog-file-format.md

```diff
@@ -62,4 +62,55 @@ Version 2
         size            : size of all entries in this multi-entry + headers
         crc             : CRC32 of magic, size
         <entries> * N
+        crc2            : CRC32 of magic, size and data in each entry
+```
+
+Version 3
+---------
+
+Modified from v3 to improve error detection/false positive elimination.
+This version does CRC per written disk block instead of actual entry.
+Every block is also tagged with which file is being written, this to
+be able to better distinguish new data from that left over from recycling
+(reusing old files) and actual disk/file corruption.
+
+```
+
+        Segment file header
+
+        magic           : uint32_t - ('S'<<24) |('C'<< 16) | ('L' << 8) | 'C';
+        version         : uint32_t - same as descriptor
+        id              : uint64_t - same as descriptor
+        alignment       : uint32_t - disk block size
+        crc             : uint32_t - CRC32 of version, low 32 of id, high 32 of id. 
+
+        Chunk header
+
+        file_pos        : uint32_t - the file position of next chunk
+        crc             : uint32_t - CRC32 of low 32 of segment id, high 32 of id and file offset of end of this header.
+
+        Entry
+
+        size            : uint32_t - size of entry (data + full headers). Must be smaller than MAX_UINT32.
+        crc             : uint32_t - CRC32 of size
+        data            : bytes - actual entry data
+
+        Multi-entry
+
+        magic           : multi marker - 0xffffffff (MAX_UINT32)
+        size            : size of all entries in this multi-entry + headers
+        crc             : CRC32 of magic, size
+        <entries> * N
+
+        Disk block (block size = `alignment`)
+
+        0 - <bs - 12>   : Interleaved file data, i.e. the content above
+        <bs - 12>       : uint32_t - CRC32 of data, low 32 of segment id, high 32 of segment id
+        <bs - 8>        : uint64_t - same as descriptor.
```

Comment:
It's nicer if crc is at the end, and treats the segment id as just bytes, so we can just use a regular buffer crc.

Reply:
But in the end it matters very little, since it is just a buffer CRC + an additional small one. It is as it is due to alignment.

Reply:
write_le will happily write unaligned data (with no loss of efficiency).

Reply:
We write network order. :-)
I wanted to keep it aligned so implementor _can_, if wanting to, do it with plain load/store to struct. Because why not. But it is not like we keep data entries and stuff aligned, so I guess it would be fine to ignore it.

Reply:
then write_be(). Using structs doesn't really help, at least during read, because we're handed unaligned memory.